### PR TITLE
Fix alamut link error

### LINKED ISSUE #4864: Error loading case general report
```
2024-09-18 07:57:41,266 - [scout.server.app](http://scout.server.app/) - ERROR: Exception on /cust004/18701400006fam/case_report [GET] [in /venv/lib/python3.11/site-packages/flask/app.py:838]
Traceback (most recent call last):
  File "/venv/lib/python3.11/site-packages/flask/app.py", line 1473, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/flask/app.py", line 882, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/flask_cors/extension.py", line 194, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
                                                ^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/flask/app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/flask/app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/worker/app/scout/server/utils.py", line 87, in decorated_function
    context = func(*args, **kwargs)
              ^^^^^^^^^^^^^^^^^^^^^
  File "/home/worker/app/scout/server/blueprints/cases/views.py", line 265, in case_report
    data = controllers.case_report_content(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/worker/app/scout/server/blueprints/cases/controllers.py", line 709, in case_report_content
    case_report_variants(store, case_obj, institute_obj, data)
  File "/home/worker/app/scout/server/blueprints/cases/controllers.py", line 630, in case_report_variants
    _append_evaluated_variant_by_type(
  File "/home/worker/app/scout/server/blueprints/cases/controllers.py", line 663, in _append_evaluated_variant_by_type
    _get_decorated_var(var_obj=var_obj, institute_obj=institute_obj, case_obj=case_obj)
  File "/home/worker/app/scout/server/blueprints/cases/controllers.py", line 639, in _get_decorated_var
    return variant_decorator(
           ^^^^^^^^^^^^^^^^^^
  File "/home/worker/app/scout/server/blueprints/variant/controllers.py", line 235, in variant
    add_gene_info(store, variant_obj, gene_panels=panels, genome_build=genome_build)
  File "/home/worker/app/scout/server/blueprints/variant/utils.py", line 264, in add_gene_info
    add_gene_links(variant_gene, genome_build, institute=institute)
  File "/home/worker/app/scout/server/links.py", line 85, in add_gene_links
    gene_obj["alamut_link"] = alamut_gene_link(institute, gene_obj, build)
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/worker/app/scout/server/links.py", line 844, in alamut_gene_link
    return url_template.format(
           ^^^^^^^^^^^^^^^^^^^^
KeyError: 'canonical_transcript'

```

### PR DESCRIPTION
**Pull request by northwestwitch**
This PR adds a functionality or fixes a bug.
- Fix #4864

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. Go to the case general report page of [this case](https://scout-stage.scilifelab.se/cust004/18701400006fam)

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DNIL
- [x] tests executed by CR

### CODE REVIEW

File: scout/server/links.py [hunk 1]
```diff
@@ -845,7 +854,7 @@ def alamut_gene_link(
         search_verb=search_verb,
         alamut_key_arg=alamut_key_arg,
         alamut_inst_arg=alamut_inst_arg,
-        this=gene_obj,
+        canonical_treanscript=get_gene_canonical_tx(gene=gene_obj),
```
Comment (dnil) [inline scout/server/links.py:857]:
```suggestion
        canonical_transcript=get_gene_canonical_tx(gene=gene_obj),
```

File: scout/server/links.py [hunk 2]
```diff
@@ -828,7 +837,7 @@ def alamut_gene_link(
     (search_verb, alamut_key_arg, alamut_inst_arg) = _get_alamut_config(institute_obj)
 
     url_template = (
-        "http://localhost:10000/{search_verb}?{alamut_key_arg}{alamut_inst_arg}request={this[canonical_transcript]}{build_str}:"
+        "http://localhost:10000/{search_verb}?{alamut_key_arg}{alamut_inst_arg}request={canonical_treanscript}{build_str}:"
```
Comment (dnil) [inline scout/server/links.py:840]:
```suggestion
        "http://localhost:10000/{search_verb}?{alamut_key_arg}{alamut_inst_arg}request={canonical_transcript}{build_str}:"
```

File: scout/server/links.py (hunk 1 above)
Comment (dnil) [reply scout/server/links.py:857]: Nice replacing the last "this"!

File: scout/server/links.py [hunk 3]
```diff
@@ -819,6 +819,15 @@ def alamut_gene_link(
         url_template(str): link to Alamut browser
     """
 
+    def get_gene_canonical_tx(gene: dict) -> str:
```
Comment (dnil) [inline scout/server/links.py:822]: This is good and should be solving the issue! Do you remember or have you found out why the issue occurs? We seem to have this kind of fallback construction in several places, either with dual checking on gene and transcript or simply falling back to checking the transcripts directly. Is it historical, or are we missing something in build?

**Review by dnil — APPROVED** [review]
This looks like it would solve the issue at hand! It is a bit odd that we have to resort to this. There are other similar constructions, so not unexpected, but would hint at an underlying issue (or backwards compatibility I didn't find on a cursory look 😸).

File: CHANGELOG.md [hunk 4]
```diff
@@ -12,7 +12,7 @@ About changelog [here](https://keepachangelog.com/en/1.0.0/)
 - `Mean MT coverage`, `Mean chrom 14 coverage` and `Estimated mtDNA copy number` on MT coverage file from chanjo2 if available
 ### Changed
 - Documentation for OMICS variants and updating a case
-- Include both creation and last modification dates in gene panels pages
+- Include both creation and deletion dates in gene panels pages
```
[the remark this answers is not in the document]
Comment (northwestwitch) [reply CHANGELOG.md:15]: Actually what is shown on that page is only creation and deletion (for deleted panels), that's why:

image

**Comment by northwestwitch** [comment]
> d, but would hint at an underlying issue (or backwards compatibility I didn't find on a cursory look 😸).

It's weird because some variant gene have the key and some don't.  this is an oldish case so perhaps some old bug that now is fixed?

File: CHANGELOG.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (northwestwitch) [reply CHANGELOG.md:15]: But whenever you modify a panel you create a new one. so it shouldn't be required no?

[the remark this answers is not in the document]
Comment (dnil) [reply CHANGELOG.md:15]: Well, creation of the version in that case. To the user the panel may have existed before they modified it. 😺

[the remark this answers is not in the document]
Comment (northwestwitch) [reply CHANGELOG.md:15]: It's clear I think, especially when you open the panel history

[the remark this answers is not in the document]
Comment (dnil) [reply CHANGELOG.md:15]: Yes, the page is reasonably clear now.

**Pull request merged**